README.md

```diff
@@ -38,229 +88,412 @@
-                        which is later can be used for building without the NodeJS
-                        runtime.
+4. Test the compiled executables inside the project's environment:
+    ```
+    % esy ./_build/default/bin/hello.exe
+    ```
 
-                        Unsupported form: build-eject [cygwin | linux | darwin]
-                        Ejects a build for the specific platform. This
-                        build-eject form is not officially supported and will
-                        be removed soon. It is currently here for debugging
-                        purposes.
+5. Hack on project's source code and rebuild the project:
+    ```
+    % esy build
+    ```
 
+Also:
 
-  import-opam           Read a provided opam file and print esy-enabled
-                        package.json conents on stdout. Example:
+6. It is possible to invoke any command from within the project's sandbox.
+   For example build & run tests with:
+    ```
+    % esy make test
+    ```
+   You can run any command command inside the project environment by just
+   prefixing it with `esy`:
+    ```
+    % esy <anycommand>
+    ```
 
-                          esy import-opam lwt 3.0.0 ./opam
+7. To shell into the project's sandbox:
+    ```
+    % esy shell
+    ```
 
-  <command>             Executes <command> as if you had executed it inside of
-                        esy shell.
+8. For more options:
+    ```
+    % esy help
+    ```
+
+### Configuring Your `package.json`
+
+`esy` knows how to build your package and its dependencies by looking at the
+`esy` config section in your `package.json`.
+
+This is how it looks for a [jbuilder][] based project:
 
 ```
+{
+  "name": "example-package",
+  "version": "1.0.0",
 
-## Try An Example
+  "esy": {
+    "build": [
+      "jbuilder build bin/hello.exe"
+    ],
+    "install": [
+      "jbuilder build @install",
+      "jbuilder install --prefix=$cur__install"
+    ],
+    "buildsinsource": "_build"
+  },
 
+  "dependencies": {
+    "anotherpackage": "1.0.0"
+  }
+}
 ```
-# Make sure esy is installed
-npm install -g esy
 
-# Clone the example esy project
-git clone git@github.com:esy-ocaml/esy-ocaml-project.git
+#### Specify Build & Install Commands
 
-cd esy-ocaml-project
+The crucial pieces of configuration are `esy.build` and `esy.install` keys, then
+specify how to build and install built artifacts.
 
-# Install project dependencies
-esy install
+##### `esy.build`
 
-# Build project
-esy build
+Describe how your project's default targets should be built by specifying
+a list of commands with `esy.build` config key.
 
-# Now run some commands inside the project's environment
-esy ./_install/bin/hello.native
+For example for a [jbuilder][] based project you'd want to call `jbuilder build`
+command for each of the target which is going to be installed later:
 
-# Shell into project's environment
-esy shell
 ```
+{
+  "esy": {
+    "build": [
+      "jbuilder build bin/hello.exe",
```

Comment:
Is the reason why we don't encourage the simpler `jbuilder build` command because it also computes the `@install` which consumes time? It would make the explanation much simpler because then the config would merely become:

```
"esy": {
  "build": ["jbuilder build"]
  ],
  "install": ["esy-installer"]
}
```

It's not *quite* as fast by default, but it's so easy to learn. Could we have an advanced sections/tips that tell you how to shave that final remaining milliseconds?

Reply:
Agree, good point.